### PR TITLE
clean flow, add WebhookMutator

### PR DESCRIPTION
**Pull request by AntonPuko**

### CODE REVIEW

File: src/lib/controllers/Controller.js [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+    status: 400,
+  });
 
-  ok(output) {
-    return {
-      data: output,
-      status: 200,
-    };
-  }
+  ok = (output?: Object): Object => ({
```
Comment (AntonPuko) [inline src/lib/controllers/Controller.js]: can't figure out the best way to annotate return type.. I was thinking about Ok<TModel> or something like that, but this way we need to do class controller<TModel> right?

File: src/types.js [hunk 2]
```diff
@@ -7,14 +9,32 @@ export type Webhook = {
...
 
-export type WebhookRequestType = 'DELETE' | 'GET' |  'POST' | 'PUT';
+export type WebhookMutator = {
+  auth?: { Authorization: string },
+  deviceID?: boolean,
+  errorResponseTopic?: string,
+  event: string,
+  form?: { [key: string]: Object },
+  headers?:{ [key: string]: string },
+  json?: { [key: string]: Object },
```
Comment (AntonPuko) [inline src/types.js]: weird type actually. form, json, query - are different ways to pass all other webhook props to post request... our create() method in Webhook controller dosen't support those now.

File: src/types.js [hunk 3]
```diff
@@ -59,12 +79,21 @@ export type UserCredentials = {
 
 export type Repository<TModel> = {
   create: (id: string, model: TModel) => TModel,
-  delete: (id: string) => void,
+  deleteById: (id: string) => void,
```
Comment (AntonPuko) [inline src/types.js]: ranamed to `deleteById` because `delete` is reserved js word.

File: src/lib/controllers/Controller.js (hunk 1 above)
Comment (jlkalberer) [inline src/lib/controllers/Controller.js]: `ok = <TType>(output?: TType): {data: ?TType, status: number} => ({`

File: src/lib/controllers/UsersController.js [hunk 4]
```diff
@@ -9,17 +12,17 @@ import httpVerb from '../decorators/httpVerb';
...
+  constructor(usersRepository: UsersRepository) {
     super();
     this._usersRepository = usersRepository;
   }
 
   @httpVerb('post')
   @route('/v1/users')
   @anonymous()
-  async createUser(userCredentials: UserCredentials) {
+  async createUser(userCredentials: UserCredentials): Promise<Object> {
```
Comment (jlkalberer) [inline src/lib/controllers/UsersController.js]: `Promise<*>`

File: src/lib/controllers/UsersController.js [hunk 5]
```diff
@@ -38,7 +41,7 @@ class UsersController extends Controller {
   @httpVerb('delete')
   @route('/v1/access_tokens/:token')
   @anonymous()
-  async deleteAccessToken(token: string) {
+  async deleteAccessToken(token: string): Promise<Object> {
```
Comment (jlkalberer) [inline src/lib/controllers/UsersController.js]: same

File: src/lib/controllers/UsersController.js [hunk 6]
```diff
@@ -54,7 +57,7 @@ class UsersController extends Controller {
   @httpVerb('get')
   @route('/v1/access_tokens')
   @anonymous()
-  async getAccessTokens() {
+  async getAccessTokens(): Promise<Object> {
```
Comment (jlkalberer) [inline src/lib/controllers/UsersController.js]: same

File: src/lib/controllers/WebhookController.js [hunk 7]
```diff
@@ -39,19 +43,19 @@ class WebhookController extends Controller {
 
   @httpVerb('get')
   @route('/v1/webhooks')
-  get() {
+  getAll(): Object {
```
Comment (jlkalberer) [inline src/lib/controllers/WebhookController.js]: `Promise<*>`

File: src/lib/controllers/WebhookController.js [hunk 8]
```diff
@@ -39,19 +43,19 @@ class WebhookController extends Controller {
...
     return this.ok(this._webhookRepository.getAll());
   }
 
   @httpVerb('get')
   @route('/v1/webhooks/:webhookId')
-  getByWebhookId(webhookId: string) {
+  getById(webhookId: string): Object {
```
Comment (jlkalberer) [inline src/lib/controllers/WebhookController.js]: same. All of these could be async functions

File: src/types.js (hunk 2 above)
Comment (jlkalberer) [inline src/types.js]: No, those are data that gets sent to the webserver (not our servers). I'm not sure what the types should be... form might just be a string.

File: test/WebhooksController.test.js [hunk 9]
```diff
@@ -136,7 +135,7 @@ test.serial('should return webhook object by id', async t => {
   t.is(testWebhook.url, response.body.url);
 });
 
-test.serial('should delete webhook', async t => {
+test.serial('should deleteById webhook', async t => {
```
Comment (jlkalberer) [inline test/WebhooksController.test.js]: nit: delete

File: src/types.js (hunk 2 above)
Comment (AntonPuko) [inline src/types.js]: ahh.. I was misunderstanding it. okey. form should be key-value object depends on docs.

Comment (jlkalberer) [inline src/types.js]: Yeah, but are you sure it's json format. key-value could be `foo=1&bar=2`

Comment (AntonPuko) [inline src/types.js]: I'm not sure, but i think for `foo=1&bar=1`  they have `query?`

**Comment by jlkalberer** [comment]
`foo=1&bar=2` can still be form data.  That's how forms are serialized but it's just sent as body content.  That can also be a query string.

**Pull request merged**